src/main/java/duke/data/task/TaskList.java

```diff
@@ -0,0 +1,148 @@
+package src.main.java.duke.data.task;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * A list of tasks that does not allow null elements or duplicates.
+ *
+ */
+public class TaskList implements Iterable<Task> {
+
+    private final List<Task> internalList = new ArrayList<>();
+
+    /**
+     * Constructs empty person list.
+     */
+    public TaskList() {}
+
+    /**
+     * Constructs a list from the items in the given collection.
+     * @param tasks a collection of persons
+     */
+    public TaskList(Collection<Task> tasks) {
+        internalList.addAll(tasks);
+    }
+
+    /**
+     * Constructs a shallow copy of the list.
+     * @param source the source of the task list
+     */
+    public TaskList(TaskList source) {
+        internalList.addAll(source.internalList);
+    }
+
+    /**
+     * Returns an unmodifiable java List view with elements cast as immutable {@link Task}s.
+     * For use with other methods/libraries.
+     * Any changes to the internal list/elements are immediately visible in the returned list.
+     * @return a list of task that is immmutable.
+     */
+    public List<Task> immutableListView() {
+        return Collections.unmodifiableList(internalList);
+    }
+
+    /**
+     * Returns an unmodifiable java List view with elements cast as immutable {@link Task}s.
+     * For use with other methods/libraries.
+     * Any changes to the internal list/elements are immediately visible in the returned list.
+     * @param searchString String to be searched in the task
+     * @return a list of tasks that is filtered
+     */
+    public List<Task> filteredView(String searchString) {
+        List<Task> filteredList = internalList
+                .stream()
+                .filter(task -> task.description.contains(searchString)).collect(Collectors.toList());
+        return Collections.unmodifiableList(filteredList);
+    }
+
+    /**
+     * Checks if the list contains an equivalent person as the given argument.
+     * defines a weaker notion of equality.
+     * @param toCheck the task to be check
+     * @return return true if the list contains the task
+     */
+    public boolean contains(Task toCheck) {
+        for (Task p : internalList) {
+            if (p == toCheck) {
+                return true;
+            }
+        }
+        return false;
+    }
```

Comment:
Boolean variables/methods should be named to sound like booleans. Perhaps changing contains(Task toCheck) method to doesContain(Task toCheck) instead?

Reply:
Changed it up. Thank you for this comment.